test/tools/yulInterpreter/Interpreter.cpp

```diff
@@ -317,14 +320,22 @@ void ExpressionEvaluator::setValue(u256 _value)
 	m_values.emplace_back(std::move(_value));
 }
 
-void ExpressionEvaluator::evaluateArgs(vector<Expression> const& _expr)
+void ExpressionEvaluator::evaluateArgs(
+	vector<Expression> const& _expr,
+	optional<vector<bool>> const& _literalArguments
+)
 {
 	vector<u256> values;
+	size_t i = 0;
 	/// Function arguments are evaluated in reverse.
 	for (auto const& expr: _expr | boost::adaptors::reversed)
 	{
-		visit(expr);
+		if (!_literalArguments || !_literalArguments->at(_expr.size() - i - 1))
```

Comment:
`!_literalArguments` here means `!literalArguments.has_value()` ?

Reply:
Looks like they are equivalent in c++17 https://en.cppreference.com/w/cpp/utility/optional/operator_bool